components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +178,31 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// calculateCompassDegreeError calculates the compass degree error
+// of two geo points.
+// GPS provides a heading data only when it has a course of direction.
+// This function provides an estimated error for that data.
+func (g *CachedData) calculateCompassDegreeError(p1, p2 *geo.Point) float64 {
+	// Check if the two positions are the same.
+	if p1.Lat() == p2.Lat() && p1.Lng() == p2.Lng() {
+		return math.NaN()
+	}
+
+	adjacent := p1.GreatCircleDistance(p2)
+
+	// If adjacent is 0, atan2 will be 90 degrees which is not desired.
```

Comment:
If `adjacent` is very close to, but not quite, 0, will we have undesirable behavior, too? I wonder if the arguments to `atan2` should be swapped, but I don't know enough to be confident of this.

Reply:
Any non-zero value is good. Arguments for atan2 is y/x, y being the radius and x being the adjacent. So no need for swap there.